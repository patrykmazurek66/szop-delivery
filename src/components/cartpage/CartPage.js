import Footer from "../footer/Footer";
import CartItemsContainer from "../cartitemscontainer/CartItemsContainer";
import "./cartpage.css";
import { Link } from "react-router-dom";
import { Button } from "@mui/material";
import { useState } from "react";

export default function CartPage() {
  const [empty, setEmpty] = useState(0);

  const handler = e => {
    setEmpty(e);
  };

  return (
    <div className="cart-page--hero">
      <div className="cart-page--cart-items-container">
        <CartItemsContainer handler={handler} />
        {empty > 0 && (
          <Link to="order">
            <Button variant="contained" fullWidth>
              Go to summary
            </Button>
          </Link>
        )}
      </div>
      <Footer />
    </div>
  );
}
