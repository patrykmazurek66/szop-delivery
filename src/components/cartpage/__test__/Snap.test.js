import renderer from "react-test-renderer";
import CartPage from "../CartPage";

it("renders correctly", () => {
  const tree = renderer.create(<CartPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
