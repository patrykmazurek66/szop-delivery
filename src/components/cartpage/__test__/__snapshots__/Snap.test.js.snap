// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="cart-page--hero"
>
  <div
    className="cart-page--cart-items-container"
  >
    <div
      className="cart-items-container"
    >
      <p
        className="cart-items-container--header"
      >
        Your Cart
      </p>
      <div
        className="cart-items-container--items-list"
      >
        <h2>
          Your cart is empty! Buy something!
        </h2>
      </div>
    </div>
  </div>
  <footer>
    <div
      className="footer-gap"
    />
    <div
      className="footer-content"
    >
      <div
        className="footer--links-container"
      >
        <a
          href="/#"
        >
          Help
        </a>
        <a
          href="/#"
        >
          FAQ
        </a>
        <a
          href="/#"
        >
          Cookies
        </a>
        <a
          href="/#"
        >
          Contact
        </a>
        <a
          href="/#"
        >
          Facebook
        </a>
        <a
          href="/#"
        >
          Log in
        </a>
        <a
          href="/#"
        >
          Sign up
        </a>
      </div>
      <p>
        Â© SzopDelivery 2022
      </p>
    </div>
  </footer>
</div>
`;
